### PR TITLE
`<barrier>`: Declare victory!

### LINKED ISSUE #3440: P2588R3 `barrier`'s Phase Completion Guarantees
WG21-P2588R3 `barrier`'s Phase Completion Guarantees

**Increased** feature-test macro (expected):
```cpp
#define __cpp_lib_barrier 202302L
```

### PR DESCRIPTION
**Pull request by AlexGuteniev**
No lines in `<barrier>` were actually touched.

I have verified that we have no test coverage on guarantees we eliminate.

Resolves #3440

### CODE REVIEW

**Comment by StephanTLavavej** [comment]
`std/language.support/support.limits/support.limits.general/barrier.version.compile.pass.cpp` is failing, otherwise this looks good to me.

File: tests/libcxx/expected_results.txt [hunk 1]
```diff
@@ -561,6 +561,9 @@ std/language.support/support.limits/support.limits.general/algorithm.version.com
 std/language.support/support.limits/support.limits.general/functional.version.compile.pass.cpp FAIL
 std/language.support/support.limits/support.limits.general/iterator.version.compile.pass.cpp FAIL
 
+# libc++ doesn't implement P2588R3 barrier's Phase Completion Guarantees
+std/language.support/support.limits/support.limits.general/barrier.version.compile.pass.cpp
```
Comment (JMazurkiewicz) [inline tests/libcxx/expected_results.txt:565]:
```suggestion
std/language.support/support.limits/support.limits.general/barrier.version.compile.pass.cpp FAIL
```

File: tests/libcxx/expected_results.txt [hunk 2]
```diff
@@ -561,6 +561,9 @@ std/language.support/support.limits/support.limits.general/algorithm.version.com
 std/language.support/support.limits/support.limits.general/functional.version.compile.pass.cpp FAIL
 std/language.support/support.limits/support.limits.general/iterator.version.compile.pass.cpp FAIL
 
+# libc++ doesn't implement P2588R3 barrier's Phase Completion Guarantees
```
Comment (CaseyCarter) [inline tests/libcxx/expected_results.txt]: Can we please move this up into the first "issues upstream knows about" section? Upstream knows they need to implement proposals that have merged into the working draft. (Ditto for `skipped_tests.txt`, of course.)

**Review by CaseyCarter — APPROVED** [review]
I'll push a change for my comment.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
I've resolved trivial adjacent-add conflicts in `yvals_core.h` and libcxx's `expected_results.txt` and `skipped_tests.txt`.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
:1st_place_medal: :heart_eyes_cat: :tada:
===

**Pull request merged**